### PR TITLE
Change minItems and maxItems to work even if those keywords are not set in the underlying schema

### PR DESCRIPTION
**Pull request by publicus**
First, thank you for your work on this project!

@pateketrueke, this PR follows our [chat conversation](https://gitter.im/json-schema-faker/Lobby?at=5de6b7bf550662459860e37b) today about the behavior of `maxItems` and `minItems`. Specifically, this PR changes JSF to use `maxItems` and `minItems` arguments passed to it (JSF), even if those keywords are not present in the underlying schema. I needed this for my own work to produce example output where both `maxItems` and `minItems` are set to `1` in JSF's arguments, but where neither keyword is present in the underlying schema; and thought to contribute the change back upstream.

Looking over the changed files, I realize that `prettier`, which is installed in my editor, made some inadvertent formatting changes as I edited. I can revert those -- before I do, though, I thought to just check in and ask whether this is a PR that you'd consider. Are you open to this idea?

Again, thank you for your work!

### CODE REVIEW

**Comment by publicus** [comment]
> Please just revert changes made by prettier, I think will be enough.

Ok! I'll work on this today, and will comment back here when it's ready for re-review!

> I liked your idea, also it can be applied on most keywords as default behavior: if such keyword is given as a global option, then it'll be applied as the default, unless it's specified (?)

Agreed!

**Comment by publicus** [comment]
@pateketrueke, this should be ready for another look! It looks much cleaner now, ha, with the formatting changes reverted. 

To confirm, the tests from `npm run dev` all still pass.

File: src/lib/types/array.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ function arrayType(value, path, resolve, traverseCallback) {
   const optionalsProbability = optionAPI('alwaysFakeOptionals') === true ? 1.0 : optionAPI('optionalsProbability');
   const fixedProbabilities = optionAPI('alwaysFakeOptionals') || optionAPI('fixedProbabilities') || false;
 
-  let length = random.number(minItems, maxItems, 1, 5);
```
Comment (pateketrueke) [inline src/lib/types/array.js]: Why this change?

Comment (publicus) [reply src/lib/types/array.js]: This ensures that if `alwaysFakeOptionals` is set, the array length is always at least 1 -- if a user set `minItems` as 0 but also set `alwaysFakeOptionals`, this would preference `alwaysFakeOptionals`. Does that make more sense,  as I'm putting it?

Comment (pateketrueke) [reply src/lib/types/array.js]: Not sure, if you revert this change, then the tests you modified are not passing due this?

Comment (publicus) [reply src/lib/types/array.js]: After reading your feedback, I realized that I *can* revert the tests I changed(!) -- I've done so just now in 15f69208. However, without this change, one test fails:

```
  203 passing (14s)
  1 failing

  1) will use alwaysFakeOptionals (tests/schema/core/issues/issue-494.json)
       should work if enabled:
     Error: account-response.current should not be empty
```

With this change in place, all tests pass for me.

Comment (pateketrueke) [reply src/lib/types/array.js]: Now I see, these lines are affecting the whole thing:

https://github.com/json-schema-faker/json-schema-faker/pull/543/files#diff-46bf7bc97cebdac6a541eb79a0ed3cd6R61-R62

The original logic below is meant to fix that kind of stuff, related with minItems and maxItems, so could you review such thing?

Comment (publicus) [reply src/lib/types/array.js]: > so could you review such thing

Sure, can do. I'll revise the approach, and re-comment here once it's ready for another look.

**Comment by publicus** [comment]
@pateketrueke, I just pushed a new set of commits that refactors the approach to be more in line with the existing logic. 

There's one change I want to ask you about: [previously](https://github.com/json-schema-faker/json-schema-faker/pull/543/commits/06626c44ace923e4a45bc95f37ad6c1064a9f7f0#diff-46bf7bc97cebdac6a541eb79a0ed3cd6L90), if `maxItems` was set in the schema, it looks like JSF would _always_ use that `maxItems` length if `optionalsProbability` is set. The change I've made switches that so that JSF will randomize the length. This would require changing the tests (there are three affected tests) that expect an exact number of array items.

This new change seems necessary, as I'm reading it, for a user to be able to set `minItems` and `maxItems` as arguments and have an array length be in that range, rather than always at `maxItems`. How does this seem to you? Are you comfortable with that change?

**Comment by pateketrueke** [comment]
I see, ideally tests related with `optionalsProbability` should not be affected, because they must be deterministic.

Also I found that just adding this code does the trick:

```diff
  if (optionAPI('maxItems')) {
+    maxItems = !minItems
+      ? optionAPI('maxItems')
+      : Math.min(optionAPI('maxItems'), minItems);

    // Don't allow user to set max items above our maximum
    if (maxItems && maxItems > optionAPI('maxItems')) {
      maxItems = optionAPI('maxItems');
    }

    // Don't allow user to set min items above our maximum
    if (minItems && minItems > optionAPI('maxItems')) {
      minItems = maxItems;
    }
  }
```

Please apply this change and add a .json test to cover this exact case, thank you!

**Comment by publicus** [comment]
> Also I found that just adding this code does the trick

Ah, indeed, thank you! I've implemented that, and added two new tests, as of 130e908. Is that all in line with what you had in mind?
